src/nvim/ex_docmd.c

```diff
@@ -9625,20 +9625,40 @@ static void ex_folddo(exarg_T *eap)
 static void ex_terminal(exarg_T *eap)
 {
   char *name = (char *)p_sh;  // Default to 'shell' if {cmd} is not given.
+  char ex_cmd[1024] = { 0 };
   bool mustfree = false;
-  char *lquote = "['";
-  char *rquote = "']";
 
+  // We will call termopen() with shell as an arglist if not given a {cmd}.
   if (*eap->arg != NUL) {
     name = (char *)vim_strsave_escaped(eap->arg, (char_u *)"\"\\");
     mustfree = true;
-    lquote = rquote = "\"";
+    snprintf(ex_cmd, sizeof(ex_cmd),
+             ":enew%s | call termopen(\"%s\") | startinsert",
+             eap->forceit ? "!" : "", name);
+  } else {
+    char **argv = build_argv((char *)p_sh);
```

Comment:
It looks like you enter a command you will get a message: `execvp failed: No such file or directory: ''`. I think people should really expect an error if they set `shell=''` and try to run the terminal.